### PR TITLE
feat(financial-aid): Web Osk features

### PR DESCRIPTION
**Pull request by MargretFinnboga**
# Web Osk features

https://app.asana.com/0/1200430512612391/board
https://app.asana.com/0/1200430512612391/1200289746691444
https://app.asana.com/0/1200430512612391/1200259492500328

## What

Web osk clean up & two endpoints added

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/financial-aid/web-osk/src/routes/application/SummaryForm/summaryForm.tsx [hunk 1]
```diff
@@ -28,27 +30,44 @@ import cn from 'classnames'
...
 } from '@island.is/financial-aid/shared'
 
+import dateAmounts from '@island.is/financial-aid-web/osk/src/utils/dateAmounts.json'
+
 interface MunicipalityData {
   municipality: Municipality
 }
 
 const SummaryForm = () => {
+  function getProperty<T, K extends keyof T>(obj: T, key: K): T[K] {
```
Comment (addi) [inline apps/financial-aid/web-osk/src/routes/application/SummaryForm/summaryForm.tsx:61]: I don't get it, what is this?

File: apps/financial-aid/web-osk/src/utils/dateAmounts.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (addi) [inline apps/financial-aid/web-osk/src/utils/dateAmounts.json:1]: This file should have a better name. TaxInfo.json?

File: apps/financial-aid/web-osk/src/utils/taxCalculator.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
+// import { AidDateAmountListItem } from './veitaFinancialAidDefinitions'
+
+export interface AidDateAmount {
+  personalTaxAllowance: number
+  taxPercentage: number
+}
+
+export const calculateAidFinalAmount = (
+  amount: number,
+  usePersonalTaxAllowance: boolean | undefined,
```
Comment (addi) [inline apps/financial-aid/web-osk/src/utils/taxCalculator.ts:10]: This function should not take it undefined, the undefined should be unwrapped out side of this function.

File: apps/financial-aid/web-osk/src/utils/taxCalculator.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
+// import { AidDateAmountListItem } from './veitaFinancialAidDefinitions'
+
+export interface AidDateAmount {
```
Comment (addi) [inline apps/financial-aid/web-osk/src/utils/taxCalculator.ts:3]: Bad name.

File: apps/financial-aid/web-osk/src/utils/useFormNavigation.ts [hunk 5]
```diff
@@ -1,92 +1,116 @@
...
+      currBranch.children &&
+      currBranch.children[obj.activeSubSectionIndex - 1]
+    ) {
+      return getPrevUrl(currBranch.children[obj.activeSubSectionIndex - 1])
+    }
+
+    return getPrevUrl(navigationTree[obj.activeSectionIndex - 1])
+  }
+
+  const bla = (navigationTree: FormStepperSection[]) => {
```
Comment (addi) [inline apps/financial-aid/web-osk/src/utils/useFormNavigation.ts:91]: Bad name.

File: apps/financial-aid/web-osk/src/utils/useFormNavigationSEC.ts [hunk 6]
```diff
@@ -0,0 +1,99 @@
+import { useEffect, useState, useContext } from 'react'
```
Comment (addi) [inline apps/financial-aid/web-osk/src/utils/useFormNavigationSEC.ts:1]: Delete this file.

File: apps/financial-aid/web-osk/src/utils/useNavigationTree.ts [hunk 7]
```diff
@@ -1,6 +1,4 @@
-import { useEffect, useState } from 'react'
-
-const useNavigationTree = () => {
+const useNavigationTree = (hasIncome: boolean | undefined) => {
```
Comment (addi) [inline apps/financial-aid/web-osk/src/utils/useNavigationTree.ts:1]: This should not take in undefined. Handle it out side of this.

File: libs/financial-aid/shared/src/lib/types.ts [hunk 8]
```diff
@@ -14,6 +14,13 @@ export enum Employment {
   OTHER = 'Other',
 }
 
+export enum State {
```
Comment (addi) [inline libs/financial-aid/shared/src/lib/types.ts:17]: This name is too generic. ApplicationState?

File: libs/financial-aid/shared/src/lib/types.ts [hunk 9]
```diff
@@ -82,6 +100,12 @@ export interface User {
   phoneNumber: string
 }
 
+export interface UsStaffer {
```
Comment (addi) [inline libs/financial-aid/shared/src/lib/types.ts:103]: What is this?

**Review by addi — CHANGES_REQUESTED** [review]

**Comment by addi** [comment]
The api does not compile, try running `yarn ts-node -P apps/financial-aid/api/tsconfig.json scripts/build-graphql-schema.ts apps/financial-aid/api/src/app/app.module`

File: apps/financial-aid/backend/migrations/20210608094446-applicaitons-add-columns-form_comment.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
+'use strict'
+
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.transaction((t) =>
+      Promise.all([
+        queryInterface.addColumn(
+          'applications',
+          'form_comment',
```
[the remark this answers is not in the document]
Comment (MargretFinnboga) [reply apps/financial-aid/backend/migrations/20210608094446-applicaitons-add-columns-form_comment.js:9]: I wanted to be specific that this is a comment that was create in the form process, if there will be more comments f.ex. in veitu, but I open for changing it

File: apps/financial-aid/web-osk/src/utils/useNavigationTree.ts (hunk 7 above)
Comment (MargretFinnboga) [reply apps/financial-aid/web-osk/src/utils/useNavigationTree.ts:1]: hasIncome is undefined until reach upon the site where you answer your income situation, should hasIncome be default false then?

File: apps/financial-aid/backend/migrations/20210608094446-applicaitons-add-columns-form_comment.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (addi) [reply apps/financial-aid/backend/migrations/20210608094446-applicaitons-add-columns-form_comment.js:9]: i see, ok. Makes sense.

File: apps/financial-aid/web-osk/src/utils/useNavigationTree.ts (hunk 7 above)
Comment (addi) [reply apps/financial-aid/web-osk/src/utils/useNavigationTree.ts:1]: The current implementation gives the same result for false and undefined. We just don't want this complexity to bleed into this function. You can just use Boolean(form.hasIncome) when you use `useNavigationTree`.

Comment (addi) [reply apps/financial-aid/web-osk/src/utils/useNavigationTree.ts:1]: Also the `FormStepperSection` type should be part of this file, not the `useFormNavigation`.

File: apps/financial-aid/web-osk/src/routes/application/EmploymentForm/employmentForm.tsx [hunk 11]
```diff
@@ -24,20 +24,47 @@ const EmploymentForm = () => {
...
-        value: item,
-      }
-    },
-  )
+  const options = [
+    Employment.WORKING,
+    Employment.UNEMPLOYED,
+    Employment.CANNOTWORK,
+    Employment.OTHER,
+  ].map((item) => {
```
[the remark this answers is not in the document]
Comment (MargretFinnboga) [reply apps/financial-aid/web-osk/src/routes/application/EmploymentForm/employmentForm.tsx:39]: I get error when I remove the return

[the remark this answers is not in the document]
Comment (addi) [reply apps/financial-aid/web-osk/src/routes/application/EmploymentForm/employmentForm.tsx:39]: Sorry it should be 
```
const options = [
    Employment.WORKING,
    Employment.UNEMPLOYED,
    Employment.CANNOTWORK,
    Employment.OTHER,
  ].map((item) => ({
    label: getEmploymentStatus[item],
    value: item,
  }))
```

[the remark this answers is not in the document]
Comment (MargretFinnboga) [reply apps/financial-aid/web-osk/src/routes/application/EmploymentForm/employmentForm.tsx]: you are not forgiven...

[the remark this answers is not in the document]
Comment (MargretFinnboga) [reply apps/financial-aid/web-osk/src/routes/application/EmploymentForm/employmentForm.tsx]: just kidding, fixed ;)

**Review by addi — APPROVED** [review]

**Pull request merged**